pool.go

```diff
@@ -298,31 +372,42 @@ func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid) (b blocks.Bl
 		}
 	}
 
-	resp, err := p.config.Client.Do(req)
+	resp, err := p.config.SaturnClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, ErrBackendFailed
+	}
+
 	fb = time.Now()
```

Comment:
We don't want to fake metrics in `goLogger.Debugw("fetch result",`  – surely producing  5XX will take more than 0ms, especially if they timeouted DHT lookup, no?  :upside_down_face: 

```suggestion
	fb = time.Now()
	if resp.StatusCode != http.StatusOK {
		return nil, ErrBackendFailed
	}


```